roxie/roxiemem/roxiemem.cpp

```diff
@@ -2330,19 +2367,17 @@ class CHeap : public CInterface
     friend class HeapCompactState;
 public:
     CHeap(CChunkingRowManager * _rowManager, const IContextLogger &_logctx, const IRowAllocatorCache *_allocatorCache, unsigned _flags)
-        : logctx(_logctx), rowManager(_rowManager), allocatorCache(_allocatorCache), activeHeaplet(NULL), heaplets(NULL), flags(_flags)
+        : logctx(_logctx), rowManager(_rowManager), allocatorCache(_allocatorCache), activeHeaplet(NULL), heaplets(NULL), flags(_flags), headMaybeSpace(BLOCKLIST_NULL)
```

Comment:
I prefer member variable initializers for the reasons you mention, and I think we should use for new code. I don't have strong feelings about changing existing code - I wouldn't object to a PR that did so but neither would I object to one that did not.